src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Contracts/ResultMessage.cs

```diff
@@ -34,6 +34,11 @@ public class ResultMessage
         /// </summary>
         public string Message { get; set; }
 
+        /// <summary>
+        /// Whether or not this message indicates the completion of a statement
+        /// </summary>
+        public bool? HasRowCount { get; set; }
```

Comment:
As mentioned in the other PR - I think it would be better to have this be `MessageType` instead corresponding to the messageType enum you made there. The row count message indicating the end of a batch is more of an implementation detail and not something our logic should need to care about.